### PR TITLE
Texture Viewer

### PR DESCRIPTION
**Pull request by DarraghBurkeMS**
Adds a "view" button to to the texture pane in the inspector, which allows you to open up a pop-out window and view the texture in fullscreen.

Use left mouse button to pan the view. Scroll wheel or +/- keys to zoom in and out.

Lays the groundwork for the texture editor.

Not sure if Node Material is a good approach or if I should rewrite the material in a more concise way.

### CODE REVIEW

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureCanvasManager.ts [hunk 1]
```diff
@@ -71,24 +67,6 @@ export class TextureCanvasManager {
 
         this._texture.updateSamplingMode(Engine.TEXTURE_NEAREST_LINEAR);
         const textureRatio = this._texture.getSize().width / this._texture.getSize().height;
-        
-        // this._backgroundTexture = new DynamicTexture("texture", 8, this._scene, false);
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureCanvasManager.ts]: We need to remove comments

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/popupComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import * as React from "react";
+import * as ReactDOM from 'react-dom';
+import { Inspector } from '../../../../../../inspector';
+
+interface IPopupComponentProps {
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/popupComponent.tsx:5]: we already have a popup component:
https://github.com/BabylonJS/Babylon.js/blob/master/inspector/src/components/actionTabs/tabs/propertyGrids/animations/popupComponent.tsx

could you move this one up so we can share it?

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureEditorMaterial.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
+import { NodeMaterial, InputBlock, NodeMaterialSystemValues, TransformBlock, VertexOutputBlock, TextureBlock, OneMinusBlock, ColorMergerBlock, AnimatedInputBlockTypes, TrigonometryBlockOperations, TrigonometryBlock, FragmentOutputBlock, FragCoordBlock, MultiplyBlock, AddBlock, ModBlock, ClampBlock, LerpBlock, Texture } from 'babylonjs';
+
+export function TextureEditorMaterial(texture : Texture) : NodeMaterial {
+    let nodeMaterial = new NodeMaterial("node");
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureEditorMaterial.ts:4]: Why not using a snippet instead? so we can get rid of this file

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureCanvasManager.ts (hunk 1 above)
Comment (DarraghBurkeMS) [reply inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureCanvasManager.ts]: Done! sorry about that

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureEditorMaterial.ts (hunk 3 above)
Comment (DarraghBurkeMS) [reply inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureEditorMaterial.ts:4]: I will look into that! Thanks

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/popupComponent.tsx (hunk 2 above)
Comment (DarraghBurkeMS) [reply inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/popupComponent.tsx:5]: Would it be okay to move the popupComponent up to /inspector/src/components? I don't want to clutter that folder but it does seem like an inspector-level component

Comment (deltakosh) [reply inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/popupComponent.tsx:5]: even better :D

File: inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureEditorMaterial.ts (hunk 3 above)
Comment (DarraghBurkeMS) [reply inspector/src/components/actionTabs/tabs/propertyGrids/materials/textures/textureEditorMaterial.ts]: Replaced it with a snippet! Thanks for the suggestion!

**Comment by deltakosh** [comment]
congrats!

**Comment by sebavan** [comment]
GG @DarraghBurkeMS !!!

**Pull request merged**